hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/LinkedTransferQueueStreamSerializer.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.serialization.impl;
+
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.internal.compatibility.serialization.impl.CompatibilitySerializationConstants;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.StreamSerializer;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+
+/**
+ * The {@code java.util.concurrent.LinkedTransferQueue} serializer
+ */
+@SuppressFBWarnings("REC_CATCH_EXCEPTION")
+public class LinkedTransferQueueStreamSerializer<E> implements StreamSerializer<Object> {
```

Comment:
Sure, adding test. And cool, I forgot about that annotation and the proposed approach, will use it.

Reply:
Copied tests for all serialisers added in 4.x. I had to use `AbstractSerializationService#toBytes` and be careful which objects are stored in the collections because actually we don't allow serialising using the "compatibility" serializers - they are meant to be used primarily for reading. Regardless, we have left the "write" code in some of them. Otherwise, the only way we could write tests would be if we copied the serialised format from 4.x tests into some kind of files.

Reply:
Oh and BTW I've kept the reflection approach. Kinda bugs me a bit whenever I use actual JDK8 classes and methods since my IDE screams at me for using it, which was what I did in the tests. I can change it to your proposed approach after more consideration and testing on JDK6 but with one slight alteration - when running on JDK6, I'd simply skip registering the serializer instead of throwing an exception.